### ISSUE TITLE
RuboCop processes files that are excluded from individual cops

### ISSUE THREAD

**Issue by rrosenblum** [bbatsov/rubocop#2142]
This was found while fixing #2119. 

If a file is excluded from an individual cop, the file is processed by the cop and then the offenses are ignored. All of the `on_` handlers are called. `investigate` will be skipped.

`Commissioner#investigate` where `process` and `investigate` are triggered from
```ruby
      def investigate(processed_source)
        reset_errors
        prepare(processed_source)
        invoke_custom_processing(@cops, processed_source)
        invoke_custom_processing(@forces, processed_source)
        process(processed_source.ast) if processed_source.ast
        @cops.each_with_object([]) do |cop, offenses|
          filename = processed_source.buffer.name
          # ignore files that are of no interest to the cop in question
          offenses.concat(cop.offenses) if cop.relevant_file?(filename)
        end 
      end
```

It would be preferable to not process the file at all for the given cop.

**Comment by bquorning** [comment]
Since `Commissioner#invoke_custom_processing` already has a `relevant_file?` check, maybe the trailing `if cop.relevant_file?(filename)` in your pasted code can just be removed? I tried it, and only this one spec failed:
```rb
    context 'when a cop has no interest in the file' do
      it 'returns all offenses except the ones of the cop' do
        cops = []
        cops << double('cop A', offenses: %w(foo), relevant_file?: true)
        cops << double('cop B', offenses: %w(bar), relevant_file?: false)
        cops << double('cop C', offenses: %w(baz), relevant_file?: true)
        cops.each(&:as_null_object)

        commissioner = described_class.new(cops, [])
        source = []
        processed_source = parse_source(source)

        expect(commissioner.investigate(processed_source)).to eq %w(foo baz)
      end
    end
```
This spec was introduced in #721 without any discussion, and seems to test this specific implementation. It seems to me that `'cop B'` would never enter the `each_with_object` loop, since the `offenses` array is populated in `#invoke_custom_processing`, and only if `relevant_file?` is true.

**Comment by bquorning** [comment]
Ah, no. The `process` call is adding offenses to the cop instances, no matter whether the files are relevant or not.

**Comment by bquorning** [comment]
Suggested a fix in #2175.
